### PR TITLE
Moving to ESLint as TSLint is now deprecated

### PR DESCRIPTION
**Pull request by adusumillipraveen**
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+    root: true,
+    parser: '@typescript-eslint/parser',
+    plugins: [
+      '@typescript-eslint',
+    ],
+    extends: [
+        'plugin:@typescript-eslint/eslint-recommended'    ],
```
Comment (timja) [inline .eslintrc.js]: Formatting looks off on this line

File: package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   },
   "scripts": {
     "start": "./node_modules/.bin/ts-node -r tsconfig-paths/register src/main/server.ts",
-    "lint": "tslint --project tsconfig.json --type-check && sass-lint -v -q",
+    "lint": "yarn eslint . --ext .js,.jsx,.ts,.tsx && sass-lint -v -q",
```
Comment (timja) [inline package.json]: Dev dependencies usually get added to the path when run as a script so eslint on its own should work

Comment (adusumillipraveen) [reply package.json]: of course, a typo :)

File: package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
...
     "test:routes": "NODE_ENV=mocha LOG_LEVEL=OFF  TS_NODE_PROJECT=tsconfig.json mocha --require ts-node/register $(find src/test \\( -name '*.ts' -or -name '*.js' \\) -path '*/routes/*') --require ts-node/register --reporter mochawesome --reporter-options reportFilename=routes,inlineAssets=true,reportTitle=expressjs-template-routes",
-    "test:a11y": "NODE_ENV=mocha LOG_LEVEL=ERROR mocha -r tsconfig-paths/register src/test/a11y --timeout 60000 --reporter mochawesome --require ts-node/register --reporter-options reportFilename=a11y,inlineAssets=true,reportTitle=expressjs-template-a11y",
+    "test:a11y": "NODE_ENV=mocha LOG_LEVEL=ERROR TS_NODE_PROJECT=tsconfig.json mocha $(find src/test \\( -name '*.ts' -or -name '*.js' \\) -path '*/a11y/*') --timeout 60000 --reporter mochawesome --require ts-node/register --reporter-options reportFilename=a11y,inlineAssets=true,reportTitle=expressjs-template-a11y",
```
Comment (timja) [inline package.json]: Isn't the tsconfig in the default location?

Comment (timja) [reply package.json]: we should improve this to have a separate folder for a11y, and just pass in the folder name, this is far more complex than should be needed

Comment (adusumillipraveen) [reply package.json]: I tried, but didn't work. wasn't spending time as i am anyway moving to jest and it might just go.

**Pull request merged**